src/app/app.py

```diff
@@ -22,15 +32,22 @@ def __init__(self, appConfig: AppConfig):
         self.logger.l.info("application started!")
 
         # setup
-        repository = appConfig.repository
-        repository_config = appConfig.repository_config
-        self.logger.l.info(
-            "creating repository -> db_type: %s, api_type: %s",
-            RepositoryEnum.to_char(repository_config.dbType),
-            RepositoryBigQueryEnum.to_char(repository_config.apiType),
-        )
-        self.repository: Repository = repository(repository_config)
-        self.logger.l.info("repository created!")
+        self.clone = appConfig.clone
+        self.clone_config = appConfig.clone_config
+
+        self.metric_config = appConfig.metric_config
+        self.metric = appConfig.metric
+
+        self.logger.l.info("cloning to github using subprocess")
+        # repository = appConfig.repository
+        # repository_config = appConfig.repository_config
+        # self.logger.l.info(
+        #     "creating repository -> db_type: %s, api_type: %s",
+        #     RepositoryEnum.to_char(repository_config.dbType),
+        #     RepositoryBigQueryEnum.to_char(repository_config.apiType),
+        # )
+        # self.repository: Repository = repository(repository_config)
+        # self.logger.l.info("repository created!")
```

Comment:
Rework needed.

Recommendations.
Create App class based on the AppConfig recommendations
e.g App, AppRepository, AppGit